homeassistant/config_entries.py

```diff
@@ -1054,7 +1054,7 @@ def async_update_entry(
         *,
         unique_id: str | None | UndefinedType = UNDEFINED,
         title: str | UndefinedType = UNDEFINED,
-        data: dict | UndefinedType = UNDEFINED,
+        data: dict | MappingProxyType | UndefinedType = UNDEFINED,
```

Comment:
Maybe we should even drop `dict` as a hint?
```python
            data: MappingProxyType | UndefinedType = UNDEFINED,
```

Reply:
I have adjusted this to `Mapping` as that covers both the `dict` and the `MappingProxyType`
I have also forced it to `[str, Any]` as that matches the type of `self.data`